src/default-markup.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * Only show comment during develop
+ */
+type Comment = (text: string) => string
+const comment: Comment = text => process.env.NODE_ENV !== 'production'
+  ? `<!--${text}-->`
+  : ``
+
+export const defaultMarkup = ({
+  title,
+  alt,
+  originSrc,
+  aspectRatio,
+  src,
+  ...props
+  // fluid: base64, srcSet, scrSetType, sizes, originalImg, density, presentationWidth, presentationHeight
+  // fixed: base64, srcSet, tracedSVG, width, height
+  // resize: absolutePath, finishedPromise, tracedSVG, width, height
+}) => {
+  const styles = {
+    fluid: {
+      imageWrapper: `
+        position: relative;
+        overflow: hidden;
+      `,
+      imagePadding: `
+        width: 100%;
+        padding-bottom: ${100 / aspectRatio}%;
+      `,
+      // temporary, would be nice to get dominate color of image passed in props
+      solidPlaceholder: `
+        background-color: #ffffff;
+        position: absolute;
+        top: 0; bottom: 0; left: 0; right: 0;
+      `,
+      // derek, you mentioned adding blur up in onClientEntry, but I'll leave this here for now:
+      imagePlaceholder: `
+        position: absolute;
+        top: 0; bottom: 0; left: 0; right: 0;
+        width: 100%;
+        height: 100%;
+        object-fit: cover;
+        object-position: center center;
+        filter: blur(50px);
+      `,
+      svgPlaceholder: `
+        position: absolute;
+        top: 0; bottom: 0; left: 0; right: 0;
+        width: 100%;
+        height: 100%;
+        object-fit: cover;
+        object-position: center center;
+      `,
+      imageTag: `
+        position: absolute;
+        top: 0; bottom: 0; left: 0; right: 0;
+        width: 100%;
+        height: 100%;
+        object-fit: cover;
+        object-position: center center;
+      `,
+    },
+    fixed: {
+      imageWrapper: `
+        position: relative;
+        overflow: hidden;
+        display: inline-block;
+        width: ${props.width || 'auto'};
+        height: ${props.height || 'auto'};
+      `,
+      solidPlaceholder: `
+        background-color: ${props.backgroundColor ||
```

Comment:
@fartinmartin any ideas if sharp actually returns `backgroundColor` and `dominateColor`? I remember seeing an issue about this in the sharp repo, but I don't know if gatsby-plugin-sharp actually return these properties 😮